src/server/replica.cc

```diff
@@ -560,16 +609,18 @@ void Replica::JoinAllFlows() {
 }
 
 void Replica::DefaultErrorHandler(const GenericError& err) {
-  CloseAllSockets();
+  CloseSocket();
 }
```

Comment:
its important we don't access the replicas in the default error handler, because if we get cancelled during the initialization phase (where it resets them) of full sync we crash if we read from them at a bad timing